src/Microsoft.AspNet.Mvc.Core/ModelBinding/CompositeModelBinder.cs

```diff
@@ -39,6 +40,12 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
                 !string.IsNullOrEmpty(bindingContext.ModelName);
 
             var newBindingContext = CreateNewBindingContext(bindingContext, bindingContext.ModelName);
+            if (newBindingContext == null)
+            {
+                // Unable to find a value provider for this binding source. Binding will fail.
+                return null;
+            }
```

Comment:
> Suggest we separate this concern into an issue about throwing for both a "missing" value provider and model binder. Thoughts?

I think I'm fine with not throwing. You're right that we don't do anything else like this in MB.